src/app/clusters/microwave-oven-control-server/microwave-oven-control-server.cpp

```diff
@@ -0,0 +1,261 @@
+/**
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ *
+ */
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app/InteractionModelEngine.h>
+#include <app/clusters/microwave-oven-control-server/microwave-oven-control-server.h>
+#include <app/clusters/mode-base-server/mode-base-server.h>
+#include <app/reporting/reporting.h>
+#include <app/util/attribute-storage.h>
+#include <app/util/error-mapping.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::OperationalState;
+using namespace chip::app::Clusters::MicrowaveOvenControl;
+using namespace chip::app::Clusters::ModeBase;
+using namespace chip::app::Clusters::MicrowaveOvenControl::Attributes;
+using Status = Protocols::InteractionModel::Status;
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace MicrowaveOvenControl {
+
+Instance::Instance(Delegate * aDelegate, EndpointId aEndpointId, ClusterId aClusterId,
+                   Clusters::OperationalState::Instance * aOpStateInstance,
+                   Clusters::ModeBase::Instance * aMicrowaveOvenModeInstance) :
+    CommandHandlerInterface(MakeOptional(aEndpointId), aClusterId),
+    AttributeAccessInterface(MakeOptional(aEndpointId), aClusterId), mDelegate(aDelegate), mEndpointId(aEndpointId),
+    mClusterId(aClusterId), mOpStateInstance(aOpStateInstance), mMicrowaveOvenModeInstance(aMicrowaveOvenModeInstance)
+{
+    mDelegate->SetInstance(this);
+}
+
+Instance::~Instance()
+{
+    InteractionModelEngine::GetInstance()->UnregisterCommandHandler(this);
+    unregisterAttributeAccessOverride(this);
+}
+
+CHIP_ERROR Instance::Init()
+{
+    // Check if the cluster has been selected in zap
+    if (!emberAfContainsServer(mEndpointId, mClusterId))
+    {
+        ChipLogError(Zcl, "Microwave Oven Control: The cluster with ID %lu was not enabled in zap.", long(mClusterId));
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+    if (!mOpStateInstance)
```

Comment:
So here is a question: Since this is a constructor argument and not mutable and not allowed to be null, should the constructor just take a reference, not pointer, the member store a reference, and then this check is not relevant (or possible) and people can't mess this up without the compiler telling them so?